File: Selnium script for testing/Script with Python/test_scenario_23.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

def execute_test_23():
    """
    Dummy test run for scenario 23
    """
    driver = webdriver.Chrome()
    driver.implicitly_wait(10)
    driver.get("https://dummy.example.com/page_23")

    driver.find_element(By.ID, "user_name").send_keys("test_data_571")
    driver.find_element(By.CSS_SELECTOR, ".promo-code").send_keys("test_data_593")
    driver.find_element(By.ID, "user_name").send_keys("test_data_416")
    driver.find_element(By.ID, "user_name").send_keys("test_data_871")
    driver.find_element(By.XPATH, "//button[@type='submit']").click()
    driver.find_element(By.ID, "checkout_btn").click()
    driver.find_element(By.LINK_TEXT, "Log Out").click()
    driver.find_element(By.LINK_TEXT, "Log Out").click()
    assert driver.find_element(By.ID, "success_msg").is_displayed()
    assert "Dashboard" in driver.title
    driver.find_element(By.XPATH, "//button[@type='submit']").click()
    driver.find_element(By.NAME, "password_field").send_keys("test_data_874")
    driver.find_element(By.CSS_SELECTOR, ".promo-code").send_keys("test_data_387")
    driver.find_element(By.XPATH, "//button[@type='submit']").click()
    driver.find_element(By.ID, "checkout_btn").click()

    driver.quit()

if __name__ == "__main__":
    execute_test_23()

File: Selnium script for testing/Script with Java/src/test/java/testscenario/TestScenario23.java
package testscenario;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import java.time.Duration;

public class TestScenario23 {
    public static void executeTest23() {
        /*
         * Dummy test run for scenario 23
         */
        WebDriver driver = new ChromeDriver();
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        driver.get("https://dummy.example.com/page_23");

        try {
            driver.findElement(By.linkText("Log Out")).sendKeys("test_data_115");
            if (!driver.findElement(By.id("checkout_btn")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            if (!driver.findElement(By.name("password_field")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.name("password_field")).sendKeys("test_data_207");
            if (!driver.findElement(By.cssSelector(".promo-code")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.cssSelector(".promo-code")).click();
            driver.findElement(By.linkText("Log Out")).sendKeys("test_data_750");
            driver.findElement(By.id("checkout_btn")).click();
            driver.findElement(By.xpath("//button[@type='submit']")).click();
            driver.findElement(By.className("nav-item")).click();
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            driver.quit();
        }
    }
    
    public static void main(String[] args) {
        executeTest23();
    }
}
